life50x50/Form1.cs

```diff
@@ -0,0 +1,206 @@
+            timer.Interval = 1000;
+            menuStrip1.Items[1].Visible = false;
+            prepgame();
+        }
+
+        private void prepgame()
+        {
+            stage = 1;
+            totalAlive = 0;
+            changes = 0;
+            toolStripStatusLabel1.Text = "Поколение "+stage.ToString();
+            for (int i = 0; i < 50; i++)
+            {
+                for (int j = 0; j < 50; j++)
+                {
+                    gameboard[i, j] = false;
+                }
+            }
+            gamestarted = false;
+
+            panel1.Refresh();
+        }
+
+        private void выходToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            Close();
+        }
+
+        private void очиститьToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            prepgame();
+        }
+
+        private void panel1_Paint(object sender, PaintEventArgs e)
+        {
+            myPen.Width = 1;
+            g = panel1.CreateGraphics();
+            for (int i = 0; i<=1000; i = i+20)
+            {
+                g.DrawLine(myPen, i, 0, i, 1000);
+                g.DrawLine(myPen, 0, i, 1000, i);
+            }
+
+            for (int i = 0; i<50; i++)
+            {
+                for (int j = 0; j<50; j++)
+                {
+                    if (gameboard [i, j] == true)
+                    {
+                        g.FillEllipse(myBrush, i * 20, j * 20, 20, 20);
+                    }
+                }
+            }
+            
+            
+        }
+
+        private void panel1_MouseClick(object sender, MouseEventArgs e)
+        {
+            if (gamestarted == false)
+            {
+                if (gameboard[e.X / 20, e.Y / 20] == false)
+                {
+                    gameboard[e.X / 20, e.Y / 20] = true;
+                    totalAlive++;
+                    panel1.Refresh();
+                }
+            }
+        }
+
+        private void стартToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            gamestarted = true;
+            menuStrip1.Items[0].Visible = false;
+            menuStrip1.Items[1].Visible = true;
+            menuStrip1.Items[2].Enabled = false;
+            timer.Start();
+        }
+
+        private void TimerEventProcessor(Object o, EventArgs e)
+        {
+            timer.Stop();
+            gameMove();
+
+            stage++;
+            toolStripStatusLabel1.Text = "Поколение " + stage.ToString();
+            panel1.Refresh();
+
+            if (changes == 0 | totalAlive == 0)
+            {
+                MessageBox.Show("Игра окончена");
+                gamestarted = false;
+                menuStrip1.Items[0].Visible = true;
+                menuStrip1.Items[1].Visible = false;
+                menuStrip1.Items[2].Enabled = true;
+            }
+            else
+            {
+                timer.Enabled = true;
+            }            

```

Comment:
И здесь уберем лишние скобки
```suggestion
            else
                timer.Enabled = true;          

```